src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-capability-modal/rgw-user-capability-modal.component.html

```diff
@@ -1,70 +1,79 @@
+        <div class="form-item">
+          <cds-text-label for="type"
+                          cdRequiredField="Type"
+                          [invalid]="formGroup.controls.type.invalid && formGroup.controls.type.dirty"
+                          [invalidText]="typeError"
+                          *ngIf="editing"
+                          i18n>Type
+            <input cdsText
+                   id="type"
+                   name="type"
+                   formControlName="type"
                    [readonly]="true"
-                   formControlName="type">
-            <select id="type"
-                    class="form-select"
-                    formControlName="type"
-                    *ngIf="!editing"
-                    autofocus>
-              <option i18n
-                      *ngIf="types !== null"
-                      [ngValue]="null">-- Select a type --</option>
-              <option *ngFor="let type of types"
-                      [value]="type">{{ type }}</option>
-            </select>
+                   modal-primary-focus>
+          </cds-text-label>
+
+          <cds-select *ngIf="!editing"
+                      label="Type"
+                      i18n-label
+                      for="type"
+                      formControlName="type"
+                      [invalid]="formGroup.controls.type.invalid && formGroup.controls.type.dirty"
+                      [invalidText]="typeError"
+                      cdRequiredField="Type">
+            <option value="">--- Select a type ---</option>
+            <option *ngFor="let type of types"
+                    [value]="type">{{ type }}</option>
+          </cds-select>
+          <ng-template #typeError>
             <span class="invalid-feedback"
                   *ngIf="formGroup.showError('type', frm, 'required')"
                   i18n>This field is required.</span>
-          </div>
+          </ng-template>
         </div>
 
         <!-- Permission -->
-        <div class="form-group row">
-          <label class="cd-col-form-label required"
-                 for="perm"
-                 i18n>Permission</label>
-          <div class="cd-col-form-input">
-            <select id="perm"
-                    class="form-select"
-                    formControlName="perm">
-              <option i18n
-                      [ngValue]="null">-- Select a permission --</option>
-              <option *ngFor="let perm of ['read', 'write', '*']"
-                      [value]="perm">
-                {{ perm }}
-              </option>
-            </select>
+        <div class="form-item">
+          <cds-select label="Permission"
+                      i18n-label
+                      for="perm"
+                      formControlName="perm"
+                      [invalid]="formGroup.controls.perm.invalid && formGroup.controls.perm.dirty"
+                      [invalidText]="permError"
+                      cdRequiredField="Permission">
```

Comment:
Same here, no need for the required
```suggestion>
```

Reply:
did this too and to other modals as well. thanks